test/indexer/test_hca_indexer.py

```diff
@@ -512,6 +512,32 @@ def mocked_mget(self, body, _source_include):
                 file_document_ids.add(file['hca_ingest']['document_id'])
         self.assertEqual(file_document_ids, file_uuids)
 
+    def test_indexing_matrix_related_files(self):
+        self._index_canned_bundle(('587d74b4-1075-4bbf-b96a-4d1ede0481b2', '2018-10-10T022343.182000Z'))
+        self.maxDiff = None
+        hits = self._get_all_hits()
+        zarrs = []
+        for hit in hits:
+            entity_type, aggregate = config.parse_es_index_name(hit["_index"])
+            if entity_type == 'files':
+                file = one(hit['_source']['contents']['files'])
+                if len(file['related_files']) > 0:
+                    self.assertEqual(file['file_format'], 'matrix')
+                    zarrs.append(hit)
+                if file['file_format'] == 'matrix':
+                    # FIXME: Why don't all matrix files have 'related_files'
```

Comment:
There are other matrix file formats like look or even csv. If it's one of these, then it's ok that there are no related files. But I would assert that.

Reply:
Interestingly, I'm not seeing these files in the test anymore. I'm not sure why this is, but I added an assert so that if there are any other matrix files with no related_files, we are sure they aren't `zarr` files.